Update affiliated package review process to delegate editorial work

astrofrog opened the pull request:
This updates the affiliated package review process to delegate editorial work to people in named roles:

https://github.com/astropy/project/issues/87

I will also open a PR to the website repo to add the new role (done: https://github.com/astropy/astropy.github.com/pull/376)

pllim commented on affiliated/affiliated_package_review_process.md:
```diff
@@ -82,41 +82,46 @@ posted on the pull request and also in a reply to your email.
-or reject) is then posted on the original pull request. If the package is
+editor, who then does their own review if needed and then writes up a review
+using the review template which includes a table with badges (template at the
+bottom of this document). At this point, the editors make a decision to access
```
Typo:
```suggestion
bottom of this document). At this point, the editors make a decision to accept
```

Maybe link to https://github.com/astropy/project/blob/master/affiliated/affiliated_package_review_guidelines.md in this section too.

astrofrog replied:
Done

pllim commented on affiliated/affiliated_package_review_process.md:
```diff
@@ -179,7 +184,7 @@ You can find our review guidelines and instructions at the following address:
   https://github.com/astropy/astropy-procedures/blob/master/documents/affiliated_package_review_guidelines.md
```
```suggestion
  https://github.com/astropy/project/blob/master/affiliated/affiliated_package_review_guidelines.md
```

astrofrog commented on affiliated/affiliated_package_review_process.md:
```diff
@@ -82,41 +82,46 @@ posted on the pull request and also in a reply to your email.
+in the project enough to be familiar with the project guidelines and requirements.
+In some cases, it may be necessary to have multiple reviewers depending on the
+expertise required, and if scientific or algorithmic expertise is needed which
+is not available in the project, reviewers can be requested from outside the
```
editors, fixed

astrofrog commented on affiliated/affiliated_package_review_process.md:
```diff
@@ -82,41 +82,46 @@ posted on the pull request and also in a reply to your email.
+is not available in the project, reviewers can be requested from outside the
+project.
+
+The editor sends out an email to possible reviewers one at a time to find
```
The one at a time meant that if you need e.g. one reviewer, not to email five and then pick the first to reply. But that's perhaps too much detail. I've changed this to 'individually' to make it clear not to send blanket emails to 5 people at the same time.

How long the editors wait for reviewers to reply is up to them.

The reviewers are anonymous unless they say they are happy to not be. However, I think @eteq and I discussed this and thought it would make sense to have it the other way around - not anonymous unless requested by the reviewer. I will make that update in a separate PR though.

astrofrog commented:
@pllim - thanks for the detailed review!

> Usage of "editor" and "editors" are inconsistent, which makes it a little confusing.

Fixed

> What if this package that is asking for review contains same functionality of another already accepted affiliated package? Will accepting both go against the "improving reuse" mantra?

This would go against the 'ecosystem integration' criterion in the guidelines, which I've now tried to clarify further. Basically we don't want to allow duplicate packages without good reason.

> I think conflict of interest cases should be clarified here too. For example, editor/reviewer knows package author very well. Hopefully, people will be honest and declare such things themselves.

I added a sentence to the email to reviewers about this.

>the text on L5 does not quite make sense to me: not the first place people should go to for - for instructions... (Why is "for" repeated? seems like an incomplete sentence before the dash.)

Fixed

> Add it at the top of the registry.json file -- I prefer alphabetical order by package name, but maybe that's just me. grimacing

I agree that would be nice, but first we then need to make sure it's actually sorted in the registry.json. Feel free to do this and then open a follow-up PR once this one is merged?


> If the package is rejected, the decision is posted to the pull request, and the pull request is closed. -- Is there a grace period before the same package can re-apply?

Not at the moment, but we could choose to add this as a guideline - but maybe in a separate PR?

> Name: Description: Repository: Documentation: -- Line break did not render in "Template email with review request"

Fixed

> Dear/hi should just be "Hi," otherwise straight copy-paste without removing one of them results in awkward email greeting.

Changed it to 'dear' as default and editors can choose to change to Hi

> Badge colors in review guidelines -- Why orange and not yellow; might as well follow the actual traffic light color scheme that it has mentioned? (See #92) Actually... I think that document needs some edits too to be consistent with the changes here.

Let's see if we can agree on this after this PR is merged?

pllim commented on affiliated/affiliated_package_review_process.md:
```diff
@@ -82,41 +82,51 @@ posted on the pull request and also in a reply to your email.
+is not available in the project, reviewers can be requested from outside the
+project at the editor's discretion.
+
+The editor sends out an email to possible reviewers, individually, to find
```
Hmm, "individually" doesn't really explain to me that you disallow blanket emails to 5 different reviewers when you only will pick one of them. I think this expectation should be its own full sentence.

astrofrog replied:
Does the latest commit clarify things? If not, can you make a suggestion?

astrofrog commented:
> I am not sure... What if the package author harasses the reviewer upon unfavorable review or something?

Ok yeah let's keep it as-is for now and if people feel strongly about the non-anonymous by default then we can open a separate PR/issue and discuss there.

pllim commented on affiliated/affiliated_package_review_process.md:
```diff
@@ -82,41 +82,53 @@ posted on the pull request and also in a reply to your email.
+The editor sends out an email to possible reviewers to find someone who is
+willing to review the package, using the template at the bottom of this
+document. Emails should only be sent to the next potential reviewer(s) - not
+many possible reviewers at the same time. The editor should let the potential
```
Something like this would be clearer for me but I don't know if that is too much. Also, you would need to break that long sentence up if you decide to adapt it here. Thanks!

```suggestion
many possible reviewers at the same time; For instance, if the review needs one reviewer, a request should first be sent to Reviewer A only and then to Reviewer B after an X amount of time has elapsed without any reply from Reviewer A (recommended X is 4 weeks and an email telling Reviewer A that their service is no longer needed should also be sent out too). The editor should let the potential
```

astrofrog replied:
Thanks! I think this is getting too specific though - I think the editors should decide on such policies internally. Also in the above scenario I think you'd want to sent reminders to reviewers etc and that really is too much detail here. My point here originally was more that the editor shouldn't just email all core maintainers asking for a volunteer, but contact people individually.

eteq commented on affiliated/affiliated_package_review_guidelines.md:
```diff
@@ -78,7 +78,7 @@ ecosystem - does it make use of existing functionality, or does it duplicate it?
 <tr>
 <td width=150><img src="https://img.shields.io/badge/Red-red.svg" alt="Red"></td>
-<td>Does not use Astropy or other affiliated packages anywhere where it should be possible, and/or uses other libraries instead.</td>
+<td>Does not use Astropy or other affiliated packages anywhere where it should be possible, and/or uses other libraries instead, or duplicates functionality found in Astropy or other affiliated packages.</td>
```
```suggestion
<td>Does not use Astropy or other affiliated packages anywhere where it should be possible, and/or uses other libraries instead, or unnecessarily duplicates functionality found in Astropy or other affiliated packages.</td>
```

eteq commented on affiliated/affiliated_package_review_process.md:
```diff
@@ -82,41 +82,53 @@ posted on the pull request and also in a reply to your email.
-then leaves a message on the pull request saying:
+*These instructions are for editors and reviewers.*
+
+The review process will be overseen and coordinated by individuals that are in
```
```suggestion
The review process will be overseen and coordinated by the individuals that are in
```

eteq commented on affiliated/affiliated_package_review_process.md:
```diff
@@ -82,41 +82,53 @@ posted on the pull request and also in a reply to your email.
+the *Affiliated package review editor* role. This role
+may be filled by one or more individuals. For the rest of this section, we
+use the singular term *editor*, although multiple people may share this role
+(and choose how to do this). The
```
```suggestion
(and it is at their discretion to decide how to divide up work). The
```

eteq commented on affiliated/affiliated_package_review_process.md:
```diff
@@ -82,41 +82,53 @@ posted on the pull request and also in a reply to your email.
+(and choose how to do this). The
+editor will start off by finding someone to carry out the main review of the
+package (hereafter the *reviewer*). In this sense the role of the editor is
+similar to that of a journal editor (and hence the role name). However this is
+only partially true, because here the editor may add their views to the review,
```
```suggestion
similar to that of a journal editor (and hence the role name). However this analogy
is only partially accurate, because here the editor may add their views to the review,
```

eteq commented on affiliated/affiliated_package_review_process.md:
```diff
@@ -82,41 +82,53 @@ posted on the pull request and also in a reply to your email.
+many possible reviewers at the same time. The editor should let the potential
+reviewer know at this point that the editor may also be doing a review and
+synthesizing the results into a single review. Once a reviewer accepts, the
+editor send them an email with detailed review instructions (which includes a
```
```suggestion
editor sends them an email with detailed review instructions (which includes a
```

eteq commented:
@astrofrog said he was fine with committing my suggestions and then merging, so here goes!
